CMakeLists.txt

```diff
@@ -12,6 +12,8 @@ foreach( _deprecation IN LISTS _deprecations )
     string( REGEX REPLACE " .+\$" "" _version "${_deprecation}" )
     if( ${_version} VERSION_LESS_EQUAL ${CMAKE_PROJECT_VERSION} )
         message( FATAL_ERROR "Deprecation messages found in deprecations.txt (deprecated since ${_version}, current engine version is ${CMAKE_PROJECT_VERSION})" )
```

Comment:
Good. I agree with it being a fatal error, not a warning as personally I don't look at the output of cmake unless it fails.

Reply:
It's the same for me, unfortunately. Also, unfortunately, it led to untested/unverified "last minute fixes" because the functionality errored out for a release. This is the best I could come up with. Any better ideas? I really don't know how to handle this properly. The only other thing which came to my mind, but which can't be automated, is to manually add issues to remove deprecations in the issue tracker. Everything I came up with involves major opportunities for human error.

Reply:
I did some research and deprecation seems to be manual and have a lot of steps for eg chromium, which is unfortunate. 

One thing I did think of for the warning case is to have it pop a little dialog box with a list of deprecations that's more likely to be seen. A little python script perhaps? I don't know how practical that is and it would have to not run in environments like LGTM with a cmake variable eg -DNO_DEPRECATION_WARNING_DIALOG. It also seems like you can't start a process in the background from cmake in a portable way so this may not be possible. 

And absolutely on putting them on the issue tracker. It still involves human error, but it sounds like a "start of sprint" procedure to complement the release procedure could be a good idea. That would involve checking all the deprecations listed in deprecations.txt have issues in the issue tracker and are on the board for the release version they need to be done in.

Reply:
One additional feature for the file could be that it needs an issue key, otherwise it would error out.

Reply:
Good Idea. Having to create an issue before you can add a deprecation would be enough I think.